### PR TITLE
Allow running extended test without specifying namespace

### LINKED ISSUE #11305: new prune extended test is broken
```
/data/src/github.com/openshift/origin/test/extended/images/prune.go:83
Expected
    <string>: Dry run enabled - no modifications will be made. Add --confirm to remove images
    Deleting references from image streams to images ...
    STREAM                                         IMAGE                                                                     TAGS
    extended-test-prune-images-4qyl3-7ma6q/prune   sha256:290043e5b69c000dd46eda62b9c98a390adf7a1792cf3fb48e898ec22e98292d   latest
to contain substring
    <string>: sha256:dd8ff54004de2f9a751ba297cf7982ad667df972ae6ace5323ab969ca52924c6
/data/src/github.com/openshift/origin/test/extended/images/prune.go:136
```

```
/data/src/github.com/openshift/origin/test/extended/images/prune.go:64
Expected
    <string>: Dry run enabled - no modifications will be made. Add --confirm to remove images
    Deleting references from image streams to images ...
    STREAM                                         IMAGE                                                                     TAGS
    extended-test-prune-images-4qyl3-fhadp/prune   sha256:77ff28ebf9cf5ece4536b810e0f5df6156140d94e247a25a5e46a7180f0566da   latest
to contain substring
    <string>: sha256:505efcf135b472e3683cace3c579bba7498a869d1c3e5fffe2b273bf516ad2f7
/data/src/github.com/openshift/origin/test/extended/images/prune.go:140
```

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #11305

@mfojtik @bparees fyi
@miminar ptal

### CODE REVIEW

**Comment by soltysh** [comment]
[testextended][extended:core(prune images)]

File: test/extended/images/prune.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ var _ = g.Describe("[images] prune images", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		g.By(fmt.Sprintf("give a user %s a right to prune images with %s role", oc.Username(), "system:image-pruner"))
-		err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-pruner", oc.Username()).Execute()
+		err = oc.AsAdmin().NoNamespace().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-pruner", oc.Username()).Execute()
```
Comment (mfojtik) [inline test/extended/images/prune.go]: I would prefer `SetNamespace("")` instead of having an extra method.

File: test/extended/util/cli.go [hunk 2]
```diff
@@ -239,10 +246,12 @@ func (c *CLI) Run(commands ...string) *CLI {
 		username:        c.username,
 		outputDir:       c.outputDir,
 		globalArgs: append(commands, []string{
-			fmt.Sprintf("--namespace=%s", c.Namespace()),
 			fmt.Sprintf("--config=%s", c.configPath),
 		}...),
 	}
+	if !c.noNamespace {
```
Comment (mfojtik) [inline test/extended/util/cli.go]: just check if `len(c.Namespace()) == 0`

File: test/extended/images/prune.go (hunk 1 above)
Comment (soltysh) [inline test/extended/images/prune.go]: We already have `SetNamespace` but that sets the namespace, which results in passing empty string, which means default namespace.

File: test/extended/util/cli.go (hunk 2 above)
Comment (soltysh) [inline test/extended/util/cli.go]: See above.

File: test/extended/images/prune.go (hunk 1 above)
Comment (mfojtik) [inline test/extended/images/prune.go]: got it, and we don't want to break that. I will name it `WithNoNamespace()` then but that is a minor nit

**Comment by mfojtik** [comment]
LGTM

File: test/extended/images/prune.go (hunk 1 above)
Comment (miminar) [inline test/extended/images/prune.go]: +1 for `WithNoNamespace()`

**Comment by mfojtik** [comment]
[test]

**Comment by soltysh** [comment]
[test]

**Comment by soltysh** [comment]
Flakes: https://github.com/openshift/origin/issues/11381 and https://github.com/openshift/origin/issues/11240

[test]

**Comment by soltysh** [comment]
@miminar @mfojtik I went with `WithoutNamespace` since `WithNoNamespace` sounds weird. I'll merge this on green tests.

**Comment by soltysh** [comment]
[merge]

**Comment by soltysh** [comment]
Flake #11442.

re-[merge]

**Pull request merged**